doc/source/data/monitoring-your-workload.rst

```diff
@@ -0,0 +1,80 @@
+.. _monitoring-your-workload:
+
+Monitoring Your Workload
+========================
+
+This section helps you debug and monitor the execution of your Ray Data :class:`~ray.data.Dataset` by viewing the:
+
+* Ray dashboard job detail page
+* Ray dashboard metrics
+* Ray Data logs
+
+Ray Data Dashboard
+------------------
+
+Ray Data emits Prometheus metrics in real-time while a Dataset is executing. These metrics are tagged by both dataset and operator, and are displayed in multiple views across the Ray dashboard.
+
+.. note::
+   Most metrics are only available for physical operators that use the map operation.
+
+Ray dashboard job detail page
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+For an overview of all datasets that have been running on your cluster, see the Ray Data Overview in the :ref:`job detail page <dash-jobs-view>`. This table appears once the first dataset starts executing on the cluster, and shows dataset details such as:
+
+* execution progress (measured in blocks)
+* execution state (running, failed, or finished)
+* dataset runtime
+* certain dataset-level metrics
+
+.. image:: images/data-overview-table.png
+   :align: center
+
+For a more fine-grained overview, each dataset row in the table can also be expanded to display the same details for individual operators.
+
+.. image:: images/data-overview-table-expanded.png
+   :align: center
+
+Ray dashboard metrics
+~~~~~~~~~~~~~~~~~~~~~
+
+For a time-series view of these metrics, see the Ray Data section in the :ref:`Metrics tab <dash-metrics-view>`. This section contains time-series graphs of all metrics emitted by Ray Data. Execution metrics are grouped by dataset and operator, and iteration metrics are grouped by dataset.
+
+The metrics recorded are:
+
+* Bytes spilled by objects from object store to disk
+* Bytes of objects allocated in object store
+* Bytes of objects freed in object store
+* Current total bytes of objects in object store
+* Logical CPUs allocated to dataset operators
+* Logical GPUs allocated to dataset operators
+* Bytes outputted by dataset operators
+* Rows outputted by dataset operators
+* Time spent generating blocks
+* Time user code is blocked during iteration.
+* Time spent in user code during iteration.
+
+.. image:: images/data-dashboard.png
+   :align: center
+
+To learn more about the Ray dashboard, including detailed setup instructions, see :ref:`Ray Dashboard <observability-getting-started>`.
+
+Ray Data logs
+-------------
+During execution, Ray Data periodically logs updates to `ray-data.log`.
+
+Every 5 seconds, Ray Data logs the execution progress of every operator in the dataset. For more frequent updates, set `RAY_DATA_TRACE_SCHEDULING=1` so that the progress is logged after each task is dispatched.
```

Comment:
```suggestion
Every five seconds, Ray Data logs the execution progress of every operator in the dataset. For more frequent updates, set `RAY_DATA_TRACE_SCHEDULING=1` so that the progress is logged after each task is dispatched.
```